utils/s3/client.cc

```diff
@@ -931,7 +930,18 @@ class client::do_upload_file {
         });
         // If this request fails, complete_multipart_upload() throws, the upload should then
         // be aborted in .close() method
-        co_await _client->make_request(std::move(req));
+        co_await _client->make_request(std::move(req), [](const http::reply& rep, input_stream<char>&& in_) -> future<> {
+            auto in = std::move(in_);
+            auto body = co_await util::read_entire_stream_contiguous(in);
+            auto possible_error = aws::aws_error::parse(std::move(body));
+            if (possible_error.get_error_type() != aws::aws_error_type::OK) {
+                if (possible_error.is_retryable()) {
+                    // Pretend we have networking problem to force the client to retry
```

Comment:
HTTP client retries only once. This explicit "failed, but you can retry" response from S3 server implies that the request can and should be retried several times, until it fails ... unretriably

Reply:
Right, also, it should use what they call retry strategy with exponential back off. IMO it is a matter of another PR. In addition we may want to do it for any request we issue, but it needs more investigation.